osutil/disks/disks_linux.go

```diff
@@ -345,52 +346,70 @@ func diskFromMountPointImpl(mountpoint string, opts *Options) (*disk, error) {
-		// controlled, so we want to remove that and just use the luks pattern
-		// to match the device uuid
-		// we are extra safe here since the dm name could be hypothetically user
-		// controlled via an external USB disk with LVM partition names, etc.
-		dmUUIDSafe := bytes.TrimSuffix(
-			bytes.TrimSpace(dmUUID),
-			append([]byte("-"), bytes.TrimSpace(dmName)...),
-		)
-		matches := luksUUIDPatternRe.FindSubmatch(dmUUIDSafe)
-		if len(matches) != 2 {
-			// the format of the uuid is different - different luks version
-			// maybe?
-			return nil, fmt.Errorf("cannot verify disk: partition %s does not have a valid luks uuid format: %s", d.Dev(), dmUUIDSafe)
-		}
+			// the uuid is the first and only submatch, but it is not in the same
+			// format exactly as we want to use, namely it is missing all of the "-"
+			// characters in a typical uuid, i.e. it is of the form:
+			// ae6e79de00a9406f80ee64ba7c1966bb but we want it to be like:
+			// ae6e79de-00a9-406f-80ee-64ba7c1966bb so we need to add in 4 "-"
+			// characters
+			compactUUID := string(matches[1])
+			canonicalUUID := fmt.Sprintf(
+				"%s-%s-%s-%s-%s",
+				compactUUID[0:8],
+				compactUUID[8:12],
+				compactUUID[12:16],
+				compactUUID[16:20],
+				compactUUID[20:],
+			)
+
+			// now finally, we need to use this uuid, which is the device uuid of
+			// the actual physical encrypted partition to get the path, which will
+			// be something like /dev/vda4, etc.
+			byUUIDPath := filepath.Join("/dev/disk/by-uuid", canonicalUUID)
+			props, err = udevPropertiesForName(byUUIDPath)
+			if err != nil {
+				return nil, fmt.Errorf("cannot get udev properties for encrypted partition %s: %v", byUUIDPath, err)
+			}
+		} else if fde.IsEncryptedDeviceMapperName(string(dmName)) {
+			// this is a device encrypted using FDE hooks
 
-		// the uuid is the first and only submatch, but it is not in the same
-		// format exactly as we want to use, namely it is missing all of the "-"
-		// characters in a typical uuid, i.e. it is of the form:
-		// ae6e79de00a9406f80ee64ba7c1966bb but we want it to be like:
-		// ae6e79de-00a9-406f-80ee-64ba7c1966bb so we need to add in 4 "-"
-		// characters
-		compactUUID := string(matches[1])
-		canonicalUUID := fmt.Sprintf(
-			"%s-%s-%s-%s-%s",
-			compactUUID[0:8],
-			compactUUID[8:12],
-			compactUUID[12:16],
-			compactUUID[16:20],
-			compactUUID[20:],
-		)
-
-		// now finally, we need to use this uuid, which is the device uuid of
-		// the actual physical encrypted partition to get the path, which will
-		// be something like /dev/vda4, etc.
-		byUUIDPath := filepath.Join("/dev/disk/by-uuid", canonicalUUID)
-		props, err = udevPropertiesForName(byUUIDPath)
-		if err != nil {
-			return nil, fmt.Errorf("cannot get udev properties for encrypted partition %s: %v", byUUIDPath, err)
+			// the uuid of the mapper device is the same
+			// as the partlabel
+			byUUIDPath := filepath.Join("/dev/disk/by-partuuid", string(dmUUID))
```

Comment:
these do not match, is the uuid of the mapper device the same as the Partition Label (as per the comment) or is it the same as the Partition UUID (as per the code)?